llama_index/llms/localai.py

```diff
@@ -44,13 +42,26 @@ def __init__(
     ) -> None:
         super().__init__(api_key=api_key, api_base=api_base, **kwargs)
 
-    def _get_context_window(self) -> int:
-        return self.context_window
+    @classmethod
+    def class_name(cls) -> str:
+        return "LocalAI"
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        return LLMMetadata(
+            context_window=self.context_window,
+            num_output=self.max_tokens or -1,
+            is_chat_model=self._is_chat_model,
+            is_function_calling_model=is_function_calling_model(
```

Comment:
@Disiok `openai_utils.is_function_calling_model` only works for OpenAI LLMs indexed into LlamaIndex. LocalAI doesn't use OpenAI's LLMs, they just copy their API endpoints